test/libsinsp_e2e/event_capture.cpp

```diff
@@ -24,208 +24,126 @@ limitations under the License.
+			continue;
 		}
-		if(!signaled_start) {
-			signaled_start = true;
-			{
-				std::lock_guard<std::mutex> lock(m_object_state_mutex);
-				m_capture_started = true;
+		if(next_result == SCAP_SUCCESS) {
+			if(m_dumper != nullptr) {
+				m_dumper->dump(event);
 			}
-			m_condition_started.notify_one();
+			result = handle_event(event);
+		} else if(m_leaving) {
+			break;
 		}
 	}
 
-	if(m_mode != SINSP_MODE_NODRIVER) {
-		uint32_t n_timeouts = 0;
-		while(result && !::testing::Test::HasFatalFailure()) {
-			{
-				std::scoped_lock inspector_next_lock(m_inspector_mutex);
-				next_result = get_inspector()->next(&event);
-			}
-			if(next_result == SCAP_TIMEOUT) {
-				n_timeouts++;
-
-				if(n_timeouts < m_max_timeouts) {
-					continue;
-				} else {
-					break;
-				}
-			}
-
-			if(next_result == SCAP_FILTERED_EVENT) {
-				continue;
-			}
-			if(next_result != SCAP_SUCCESS) {
-				break;
-			}
-			if(m_dump) {
-				dumper->dump(event);
+	/*
+	 * Second loop to empty the buffers from all the generated events:
+	 * * loop until SCAP_TIMEOUT is received.
+	 */
+	result = true;
+	while(result) {
+		next_result = m_inspector->next(&event);
+		if(next_result == SCAP_SUCCESS) {
+			if(m_dumper != nullptr) {
+				m_dumper->dump(event);
 			}
 			result = handle_event(event);
-		}
-		{
-			std::scoped_lock inspector_next_lock(m_inspector_mutex);
-			while(SCAP_SUCCESS == get_inspector()->next(&event)) {
-				// just consume the remaining events
-			}
+		} else if(next_result == SCAP_TIMEOUT) {
+			break;
 		}
 	}
 
-	{  // Begin teardown synchronized section
-		std::scoped_lock teardown_lock(m_inspector_mutex, m_object_state_mutex);
-		m_before_close(get_inspector());
-
-		get_inspector()->stop_capture();
-		if(m_mode != SINSP_MODE_NODRIVER) {
-			dumper->close();
-		}
-
-		m_capture_stopped = true;
-	}  // End teardown synchronized section
-
-	m_condition_stopped.notify_one();
-}
-
-void event_capture::stop_capture() {
-	{
-		std::scoped_lock init_lock(m_inspector_mutex, m_object_state_mutex);
-		get_inspector()->stop_capture();
-		m_capture_stopped = true;
-		m_condition_stopped.notify_one();
-	}
-}
-
-void event_capture::wait_for_capture_start() {
-	std::unique_lock<std::mutex> lock(m_object_state_mutex);
-	m_condition_started.wait(lock, [this]() { return m_capture_started; });
-}
-
-void event_capture::wait_for_capture_stop() {
-	std::unique_lock<std::mutex> lock(m_object_state_mutex);
-	m_condition_stopped.wait(lock, [this]() { return m_capture_stopped; });
-}
-
-void event_capture::re_read_dump_file() {
-	try {
-		sinsp inspector;
-		sinsp_evt* event;
-
-		inspector.open_savefile(m_dump_filename);
-		uint32_t res;
-		do {
-			res = inspector.next(&event);
-		} while(res == SCAP_SUCCESS);
-		ASSERT_EQ((int)SCAP_EOF, (int)res);
-	} catch(sinsp_exception& e) {
-		FAIL() << "caught exception " << e.what();
-	}
+	auto capture_stats_str = capture_stats(m_inspector.get());
+	std::cout << capture_stats_str << "\n";
```

Comment:
Always print capture_stats upon leaving so that we know if any drop was present.